### PR TITLE
[AppInsights] adding e2e tests; removing QuickPulse test as it is currently unsupported

### PR DESCRIPTION
**Pull request by brettsam**

### CODE REVIEW

File: test/Microsoft.Azure.WebJobs.Host.EndToEndTests/ApplicationInsightsEndToEndTests.cs [hunk 1]
```diff
@@ -46,272 +40,215 @@ public class ApplicationInsightsEndToEndTests
...
+                await host.StartAsync();
+                var methodInfo = GetType().GetMethod(testName, BindingFlags.Public | BindingFlags.Static);
+                await host.CallAsync(methodInfo, new { input = "function input" });
+                await host.StopAsync();
+            }
 
-                    await host.StopAsync();
-                }
+            // wait for everything to flush
+            await Task.Delay(2000);
```
Comment (mamaso) [inline test/Microsoft.Azure.WebJobs.Host.EndToEndTests/ApplicationInsightsEndToEndTests.cs]: is there a better way than the delay?

File: test/Microsoft.Azure.WebJobs.Host.EndToEndTests/ApplicationInsightsEndToEndTests.cs [hunk 2]
```diff
@@ -46,272 +40,215 @@ public class ApplicationInsightsEndToEndTests
...
+            Assert.Equal(6, _channel.Telemetries.Count);
+
+            // Validate the traces. Order by message string as the requests may come in
+            // slightly out-of-order or on different threads
+            TraceTelemetry[] telemetries = _channel.Telemetries
+                .OfType<TraceTelemetry>()
+                .OrderBy(t => t.Message)
+                .ToArray();
+
+            ValidateTrace(telemetries[0], "Found the following functions:\r\n", LogCategories.Startup);
```
Comment (mamaso) [inline test/Microsoft.Azure.WebJobs.Host.EndToEndTests/ApplicationInsightsEndToEndTests.cs]: Will these telemetry assertions be reliable?

File: test/Microsoft.Azure.WebJobs.Host.EndToEndTests/ApplicationInsightsEndToEndTests.cs (hunk 1 above)
Comment (brettsam) [inline test/Microsoft.Azure.WebJobs.Host.EndToEndTests/ApplicationInsightsEndToEndTests.cs]: Actually I don't need this anymore; I'll remove.

File: test/Microsoft.Azure.WebJobs.Host.EndToEndTests/ApplicationInsightsEndToEndTests.cs (hunk 2 above)
Comment (brettsam) [inline test/Microsoft.Azure.WebJobs.Host.EndToEndTests/ApplicationInsightsEndToEndTests.cs]: Yes -- they're ordered by the message string so even timing won't affect it.

**Comment by brettsam** [comment]
Merging this to unblock next PR. Can still revise if there are additional comments.

**Pull request merged**